src/transactions/ManageOfferOpFrame.cpp

```diff
@@ -223,10 +248,39 @@ ManageOfferOpFrame::doApply(Application& app, LedgerDelta& delta,
         Price const& sheepPrice = mSellSheepOffer->getPrice();
         const Price maxWheatPrice(sheepPrice.d, sheepPrice.n);
 
-        int64_t maxWheatReceive = canBuyAtMost(wheat, mWheatLineA);
+        int64_t maxWheatReceive =
+            canBuyAtMost(mSourceAccount, wheat, mWheatLineA, ledgerManager);
         int64_t maxSheepSend;
         if (app.getLedgerManager().getCurrentLedgerVersion() >= 10)
         {
+            int64_t availableLimit =
+                (wheat.type() == ASSET_TYPE_NATIVE)
+                    ? mSourceAccount->getMaxAmountReceive(ledgerManager)
+                    : mWheatLineA->getMaxAmountReceive(ledgerManager);
+            if (availableLimit < mSellSheepOffer->getBuyingLiabilities())
+            {
+                app.getMetrics()
+                    .NewMeter({"op-manage-offer", "invalid", "line-full"},
+                              "operation")
+                    .Mark();
+                innerResult().code(MANAGE_OFFER_LINE_FULL);
+                return false;
+            }
+
+            int64_t availableBalance =
+                (sheep.type() == ASSET_TYPE_NATIVE)
+                    ? mSourceAccount->getBalanceAboveReserve(ledgerManager)
```

Comment:
I think we could change getBalanceAboveReserver name to getAvailableBalance for symmetry?